daemon/emer-aggregate-tally.c

```diff
@@ -0,0 +1,373 @@
+                                       GVariant           *event_id,
+                                       GVariant           *aggregate_key,
+                                       const char         *date)
+{
+  g_autofree gchar *tally_path = NULL;
+  g_autofree gchar *contents = NULL;
+  g_autoptr(GVariant) variant = NULL;
+  g_autoptr(GVariant) unused_event_id = NULL;
+  g_autoptr(GVariant) unused_aggregate_key = NULL;
+  g_autoptr(GVariant) unused_payload = NULL;
+  g_autoptr(GError) error = NULL;
+  guint32 previous_counter;
+  guint32 unused_unix_user_id;
+  gsize size;
+
+  tally_path = get_tally_path_from_aggregate_timer (self,
+                                                    unix_user_id,
+                                                    event_id,
+                                                    aggregate_key,
+                                                    date);
+
+  if (!g_file_test (tally_path, G_FILE_TEST_EXISTS))
+    return 0;
+
+  g_file_get_contents (tally_path, &contents, &size, &error);
+  if (error)
+    return 0;
+
+  variant = g_variant_new_from_data (G_VARIANT_TYPE ("(vvuumv)"),
+                                     contents, size, FALSE, NULL, NULL);
+  if (error)
+    return 0;
+
+  if (!g_variant_check_format_string (variant, "(vvuumv)", FALSE))
+    return 0;
+
+  g_variant_get (variant,
+                 "(vvvuumv)",
+                 &unused_event_id,
+                 &unused_aggregate_key,
+                 &unused_unix_user_id,
+                 &previous_counter,
+                 &unused_payload);
+
+  return previous_counter;
+}
+
+static void
+emer_aggregate_tally_finalize (GObject *object)
+{
+  EmerAggregateTally *self = (EmerAggregateTally *)object;
+
+  g_clear_pointer (&self->persistent_cache_directory, g_free);
+
+  G_OBJECT_CLASS (emer_aggregate_tally_parent_class)->finalize (object);
+}
+
+static void
+emer_aggregate_tally_set_property (GObject      *object,
+                                   guint         prop_id,
+                                   const GValue *value,
+                                   GParamSpec   *pspec)
+{
+  EmerAggregateTally *self = EMER_AGGREGATE_TALLY (object);
+
+  switch (prop_id)
+    {
+    case PROP_PERSISTENT_CACHE_DIRECTORY:
+      g_assert (self->persistent_cache_directory == NULL);
+      self->persistent_cache_directory = g_value_dup_string (value);
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+    }
+}
+
+static void
+emer_aggregate_tally_class_init (EmerAggregateTallyClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->finalize = emer_aggregate_tally_finalize;
+  object_class->set_property = emer_aggregate_tally_set_property;
+
+  /*
+   * EmerDaemon:persistent-cache-directory:
+   *
+   * A directory for temporarily storing events until they are uploaded to the
+   * metrics servers. If a network send provider is not specified, a default
```

Comment:
"network send provider" – copy-pasta error?

Reply:
In my defense, EmerDaemon also has this copy-pasta!